activity_calendar/models.py

```diff
@@ -356,13 +354,17 @@ def get_absolute_url(self):
 class ActivitySlot(models.Model):
     title = models.CharField(max_length=255)
     description = models.TextField(blank=True)
-    location = models.CharField(max_length=255, blank=True, null=True,
-        help_text="If left empty, matches location with activity")
     start_date = models.DateTimeField(blank=True, null=True,
         help_text="If left empty, matches start date with activity")
     end_date = models.DateTimeField(blank=True, null=True,
         help_text="If left empty, matches end date with activity")
 
+
+    location = models.CharField(max_length=255, blank=True, null=True,
+        help_text="If left empty, matches location with activity")
+    location_is_private = models.BooleanField(default=False,
+        help_text="Private locations are hidden for users not registered to this slot", verbose_name="private location?")
+
```

Comment:
While the request indeed does request that slots should be turned invisible, I have my doubts if we want it to be configurable on the slots themselves.
It would make more sense at the activitymoment or activity level itself so that all locations are possibly hidden until after sign-in or slot-registration (whatever is more favourable). It prevents mixed communications on a single page.

Reply:
Agreed! Will move to the activity(moment) model. 👍